aws/src/test/java/org/apache/iceberg/aws/s3/S3OutputStreamTest.java

```diff
@@ -133,6 +135,64 @@ public void testAbortMultipart() {
     }
   }
 
+  public void testFailWriteIntAfterAsyncError() throws Exception {
+    doThrow(new RuntimeException()).when(s3mock).uploadPart((UploadPartRequest) any(), (RequestBody) any());
```

Comment:
nit: the code from L139-L146 is heavily repeated, we can move it to a helper private method

Reply:
good point. refactored the test to avoid duplicated code.